program/src/state.rs

```diff
@@ -76,6 +222,11 @@ pub(crate) fn collect_holder_rewards_pool_signer_seeds<'a>(
 #[derive(Clone, Copy, Debug, Default, PartialEq, Pod, ShankAccount, Zeroable)]
 #[repr(C)]
 pub struct HolderRewards {
+    /// The rewards per token exchange rate when this holder last harvested.
+    ///
+    /// Stored as a `u128`, which includes a scaling factor of `1e9` to
```

Comment:
We should definitely land this with a 1e9 scaling factor, since that "seems" reasonable, but now that we're introducing less precise math, the proptests will be even more important to figure out the exact precision of the system. For example, if I do a transfer + harvest vs harvest, what's the maximum difference we can expect between those two operations?

Not for this PR of course, but just some extra food for thought

Reply:
Cool, thanks, this is good context! Noted.